internal/gitutil/git_test.go

```diff
@@ -0,0 +1,213 @@
+	t.Helper()
+
+	ok, err := IsGitCloned(repoPath)
+
+	if err != nil {
+		t.Errorf("expected to finish correctly: %s", err)
+	}
+
+	if ok != true {
+		t.Errorf("expected to return true after cloning the repo")
+	}
+}
+
+func TestEnsureClonedClonesARepository(t *testing.T) {
+	tempDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+
+	httpClonePath := tempDir.Path("krew-from-https")
+	localClonePath := tempDir.Path("krew-from-local-path")
+
+	if err := EnsureCloned("https://github.com/kubernetes-sigs/krew.git", httpClonePath); err != nil {
+		t.Errorf("http clone expected to finish correctly: %s", err)
+	}
+
+	if err := EnsureCloned(httpClonePath, localClonePath); err != nil {
+		t.Errorf("folder clone expected to finish correctly: %s", err)
+	}
+
+	assertRepoIsCloned(t, httpClonePath)
+	assertRepoIsCloned(t, localClonePath)
+}
+
+func TestEnsureClonedDoesntCloneAnExistingRepo(t *testing.T) {
+	tempDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+
+	if err := EnsureCloned("https://github.com/kubernetes-sigs/krew.git", tempDir.Root()); err != nil {
+		t.Errorf("clone expected to finish correctly: %s", err)
+	}
+
+	tempDir.Write("file", []byte("create a file to ensure is not overwriten in the next clone"))
+
+	if err := EnsureCloned("https://github.com/kubernetes-sigs/krew.git", tempDir.Root()); err != nil {
+		t.Errorf("second clone expected to finish correctly: %s", err)
+	}
+
+	_, err := os.Stat(tempDir.Path("file"))
+
+	if err != nil {
+		t.Error("expected to dont perform any operation in the second clone")
+	}
+}
+
+func TestIsGitClonedWhenPathDoesntExists(t *testing.T) {
+	tempDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+
+	ok, err := IsGitCloned(tempDir.Path("does-not-exist"))
+
+	if err != nil {
+		t.Errorf("expected to finish correctly: %s", err)
+	}
+
+	if ok != false {
+		t.Errorf("expected to return false on a non existing folder")
+	}
+}
+
+func TestIsGitClonedWhenIsFalse(t *testing.T) {
+	tempDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+
+	ok, err := IsGitCloned(tempDir.Root())
+
+	if err != nil {
+		t.Errorf("expected to finish correctly: %s", err)
+	}
+
+	if ok != false {
+		t.Errorf("expected to return false on a folder that's not a git repo")
+	}
+}
+
+func TestIsGitClonedWhenIsTrue(t *testing.T) {
+	tempDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+
+	if err := os.MkdirAll(tempDir.Path(".git"), os.ModePerm); err != nil {
+		t.Fatalf("cannot create directory %q: %s", filepath.Dir(tempDir.Path(".git")), err)
+	}
+
+	ok, err := IsGitCloned(tempDir.Root())
+
+	if err != nil {
+		t.Errorf("expected to finish correctly: %s", err)
+	}
+
+	if ok != true {
+		t.Errorf("expected to return true on a git repo")
+	}
+}
+
+func TestEnsureUpdatedClonesTheRepoIfDoesntExists(t *testing.T) {
+	tempDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+
+	if err := EnsureUpdated("https://github.com/kubernetes-sigs/krew.git", tempDir.Root()); err != nil {
```

Comment:
since this test is using the network, it's quite bad for a unit test (which is supposed to run fast).

if possible, we should be using a local directory as the remote repo. for that, we need to init a dir, maybe add a file and make 1 or 2 commits, then clone that dir.

Reply:
That's why I was asking if I should mock the `Exec` call. to make them faster and don't depend on git and other stuff.

I can make all tests use a local git repo and only the one about https working to use the url if that sounds ok

Reply:
Mocking `Exec` should not be necessary. I think it's fine to clone a local repository (see other comment above).
Also, I don't think that cloning from a remote repo is something that we should test in unit tests.